src/librustdoc/clean/types.rs

```diff
@@ -1608,6 +1575,37 @@ impl GetDefId for Type {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, Copy, Debug)]
+/// N.B. this has to be different from `hir::PrimTy` because it also includes types that aren't
+/// paths, like `Unit`.
```

Comment:
While you're at it, and since you did this for `ItemLink`, can you move the `derive` to after the docs?

Reply:
After that, I'll take another look at the other changes, and then I think this should be good to go :)

Reply:
Yes. :)